program/src/oracle/upd_aggregate.h

```diff
@@ -205,39 +206,76 @@ static inline void upd_aggregate( pc_price_t *ptr, uint64_t slot )
   ptr->valid_slot_ = ptr->agg_.pub_slot_;// valid slot-time of agg. price
   ptr->agg_.pub_slot_ = slot;            // publish slot-time of agg. price
 
+  // identify valid quotes
+  // compute the aggregate prices and ranges
   uint32_t numv = 0;
   uint32_t vidx[ PC_COMP_SIZE ];
-  // identify valid quotes and order them by price
-  for ( uint32_t i = 0; i != ptr->num_; ++i ) {
-    pc_price_comp_t *iptr = &ptr->comp_[i];
-    // copy contributing price to aggregate snapshot
-    iptr->agg_ = iptr->latest_;
-    // add quote to sorted permutation array if it is valid
-    int64_t slot_diff = ( int64_t )slot - ( int64_t )( iptr->agg_.pub_slot_ );
-    if ( iptr->agg_.status_ == PC_STATUS_TRADING &&
-         iptr->agg_.conf_ != 0UL &&
-         iptr->agg_.price_ > 0L &&
-         slot_diff >= 0 && slot_diff <= PC_MAX_SEND_LATENCY ) {
-      vidx[numv++] = i;
+  int64_t  agg_price;
+  int64_t  agg_conf;
+  {
+    uint32_t nprcs = (uint32_t)0;
+    int64_t  prcs[ PC_COMP_SIZE * 3 ]; // ~0.75KiB for current PC_COMP_SIZE (FIXME: DOUBLE CHECK THIS FITS INTO STACK FRAME LIMIT)
+    for ( uint32_t i = 0; i != ptr->num_; ++i ) {
+      pc_price_comp_t *iptr = &ptr->comp_[i];
+      // copy contributing price to aggregate snapshot
+      iptr->agg_ = iptr->latest_;
+      // add quote to sorted permutation array if it is valid
+      int64_t slot_diff = ( int64_t )slot - ( int64_t )( iptr->agg_.pub_slot_ );
+      int64_t price     = iptr->agg_.price_;
+      int64_t conf      = ( int64_t )( iptr->agg_.conf_ );
+      if ( iptr->agg_.status_ == PC_STATUS_TRADING &&
+           (int64_t)0 < conf && conf < price && conf <= (INT64_MAX-price) && // No overflow for INT64_MAX-price as price>0
+           slot_diff >= 0 && slot_diff <= PC_MAX_SEND_LATENCY ) {
+        vidx[numv++] = i; // FIXME: GRAFT FOR OLD PRICE MODEL
+        prcs[ nprcs++ ] = price - conf; // No overflow as 0 < conf < price
+        prcs[ nprcs++ ] = price;
+        prcs[ nprcs++ ] = price + conf; // No overflow as 0 < conf <= INT64_MAX-price
+      }
+    }
+
+    // too few valid quotes
+    ptr->num_qt_ = numv; // FIXME: TEMPORARY GRAFT (ALL RETURN PATHS SET NUM_QT TO SOMETHING SENSIBLE)
```

Comment:
why is this temporary? seems like if we delete the old aggregation logic, this should still be set here?